ARRAY JOIN operations are not recognised in Clickhouse queries

Consider the following example:

```python
import os

import pytest
from sql_mock.clickhouse import column_mocks as col
from sql_mock.clickhouse.table_mocks import ClickHouseTableMock
from sql_mock.table_mocks import table_meta

QUERY = """SELECT
    sum(1) AS impressions,
    city,
    browser
FROM
(
    SELECT
        ['Istanbul', 'Berlin', 'Bobruisk'] AS cities,
        ['Firefox', 'Chrome', 'Chrome'] AS browsers
)
ARRAY JOIN
    cities AS city,
    browsers AS browser
GROUP BY
    city,
    browser
"""


@table_meta(query=QUERY)
class QueryMock(ClickHouseTableMock):
    impressions = col.Int(0)
    city = col.String("Istanbul")
    browser = col.String("Firefox")


@pytest.fixture(autouse=True)
def clickhouse_credentials() -> None:
    os.environ["SQL_MOCK_CLICKHOUSE_HOST"] = "127.0.0.1"
    os.environ["SQL_MOCK_CLICKHOUSE_PORT"] = "8123"
    os.environ["SQL_MOCK_CLICKHOUSE_USER"] = ""
    os.environ["SQL_MOCK_CLICKHOUSE_PASSWORD"] = ""


def test_query() -> None:
    expected = [
        {
            "impressions": 1,
            "city": "Istanbul",
            "browser": "Firefox",
        },
        {
            "impressions": 1,
            "city": "Berlin",
            "browser": "Chrome",
        },
        {
            "impressions": 1,
            "city": "Bobruisk",
            "browser": "Chrome",
        },
    ]
    result = QueryMock.from_mocks(input_data=[])
    result.assert_equal(expected)
```

When testing this, `sql-mock` will fail the test with the following error:

```shell
FAILED tests/test_example.py::test_query - sql_mock.exceptions.ValidationError: You need to provide the following input mocks to run your query: ['browsers', 'cities']
```
The arguments to the `ARRAY JOIN` operation should not need to be mocked as they are columns in the sub-query.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 * Add default target path for dbt
 * Improve replacement of tables (also taking into account missing alias)
+* Do not require mocks for ARRAY JOIN clause arguments
 
 ## [0.6.0]
 

--- a/src/sql_mock/helpers.py
+++ b/src/sql_mock/helpers.py
@@ -144,6 +144,10 @@ def get_source_tables(query, dialect) -> List[str]:
         #     then `source` will be a Table instance.
         for alias, (node, source) in scope.selected_sources.items()
         if isinstance(source, sqlglot.exp.Table)
+        # When using ARRAY joins sqlglot percieves the inputs as tables even though they are infact not.
+        # This fixes it but does not allow for multiple types of joins to be mixed with the ARRAY JOIN,
+        # For now we consider it a reasonable solution.
+        and not (node.parent.key == "join" and any(join.kind == "ARRAY" for join in node.parent_select.args["joins"]))
     }
 
     return list(tables)

--- a/tests/sql_mock/test_helpers.py
+++ b/tests/sql_mock/test_helpers.py
@@ -321,3 +321,27 @@ class TestGetSourceTables:
 
         expected = ["table_1"]
         assert res == expected
+
+    def test_query_with_array_joins(self):
+        """...then the array join fields should not be treated as tables to be mocked"""
+
+        query = """
+        SELECT
+            sum(1) AS impressions,
+            city,
+            browser
+        FROM
+        (
+            SELECT
+                ['Istanbul', 'Berlin', 'Bobruisk'] AS cities,
+                ['Firefox', 'Chrome', 'Chrome'] AS browsers
+        )
+        ARRAY JOIN
+            cities AS city,
+            browsers AS browser
+        """
+
+        res = get_source_tables(query, dialect="clickhouse")
+
+        expected = []
+        assert res == expected

